src/main/java/team/silvertown/masil/masil/repository/MasilQueryRepositoryImpl.java

```diff
@@ -0,0 +1,39 @@
+package team.silvertown.masil.masil.repository;
+
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import jakarta.persistence.EntityManager;
+import java.util.List;
+import java.util.Objects;
+import org.springframework.stereotype.Repository;
+import team.silvertown.masil.masil.domain.Masil;
+import team.silvertown.masil.masil.domain.QMasil;
+import team.silvertown.masil.user.domain.User;
+
+@Repository
+public class MasilQueryRepositoryImpl implements MasilQueryRepository {
+
+    private static final int DEFAULT_RECENT_SIZE = 10;
+
+    private final JPAQueryFactory jpaQueryFactory;
+
+    public MasilQueryRepositoryImpl(EntityManager entityManager) {
+        this.jpaQueryFactory = new JPAQueryFactory(entityManager);
+    }
+
+    public List<Masil> findRecent(User user, Integer size) {
+        QMasil masil = QMasil.masil;
+        int limit = DEFAULT_RECENT_SIZE;
+
+        if (Objects.nonNull(size) && size != 0) {
+            limit = size;
+        }
```

Comment:
파라미터에서 `size`를 wrapper class 말고 primitive로 받아서 아래와 같이 바꾸는 건 어때요?
컨트롤러에서는 NonNull 체크를 하고
```suggestion
    public List<Masil> findRecent(User user, Integer size) {
        QMasil masil = QMasil.masil;
        int limit = DEFAULT_RECENT_SIZE;

        if (size > 0) {
            limit = size;
        }
```

Reply:
저도 처음에 그렇게 했었는데 엔티티 필드 타입이랑 인자 타입이랑 wrapper랑 primitive도 구분이 돼야 하더라고요..안그럼 예외가 터져서 이렇게 했었어ㅛ유ㅠ

Reply:
아 그렇군요 ㅠㅠ

Reply:
query dsl 이슈 같아요